clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -178,4 +184,66 @@ public static Properties producerConfig(final String bootstrapServers,
+    public static Properties producerConfigStringKeysValues(final String bootstrapServers) {
+        return producerConfig(bootstrapServers, StringSerializer.class, StringSerializer.class, new Properties());
+    }
+
+    // provides producer properties with default serializers for Integers keys and String values
+    public static Properties producerConfigIntKeysStringValues(final String bootstrapServers) {
+        return producerConfig(bootstrapServers, IntegerSerializer.class, StringSerializer.class, new Properties());
+    }
+
+    public static Properties consumerConfig(final String bootstrapServers,
+                                            final String groupId,
+                                            final Class keyDeserializer,
+                                            final Class valueDeserializer,
+                                            final Properties additional) {
+
+        final Properties consumerConfig = new Properties();
+        consumerConfig.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+        consumerConfig.put(ConsumerConfig.GROUP_ID_CONFIG, groupId);
+        consumerConfig.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        consumerConfig.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, keyDeserializer);
+        consumerConfig.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, valueDeserializer);
+        consumerConfig.putAll(additional);
+        return consumerConfig;
+    }
+
+    // returns consumer config with random UUID for the Group ID and default StringDeserializer for keys/values
+    public static Properties consumerConfigStringKeysValues(final String bootstrapServers) {
+        return consumerConfig(bootstrapServers,
+                              UUID.randomUUID().toString(),
+                              StringDeserializer.class,
+                              StringDeserializer.class,
+                              new Properties());
+    }
+
+    // returns consumer config with random UUID for the Group ID and default serializers
+    public static Properties consumerConfigIntKeysStringValues(final String bootstrapServers) {
+        return consumerConfig(bootstrapServers,
+                UUID.randomUUID().toString(),
+                IntegerDeserializer.class,
+                StringDeserializer.class,
+                new Properties());
+    }
+
+    // uses default value of 30 seconds for timeout
+    public static void waitForCondition(TestCondition testCondition) {
+        waitForCondition(testCondition, 30000);
+    }
+
+    /**
+     *  Used to wait for specific conditions/state to be me during a test
+     *  this is meant to be a replacement for using Thread.sleep
+     */
+    public static void waitForCondition(TestCondition testCondition, long maxTimeMillis) {
+        long startTime = System.currentTimeMillis();
+        while (!testCondition.conditionMet() && ((System.currentTimeMillis() - startTime) < maxTimeMillis)) {
```

Comment:
> Consider adding some small back-off time (say 100 millis) to avoid busy looping and possible log polluting.

@guozhangwang 

Aggreed,  One question since we are trying to get way from Thread.sleep calls should do I some thing like this (very roughly)
```
waitForCondition(Condition cond) {
 Object localObject = new Object();//local object for each call shouldn't block other callers
 while (conditionNotMet && timeNotExpired) {
          synchonized(localObject){
               wait(100);
        }
   }
}
```
Or is this overkill and just go with Thread.sleep(100) ?